contracts/protocore/Protocore.sol

```diff
@@ -64,14 +64,58 @@ contract Protocore is MosaicVersion, ValidatorSet, CoconsensusModule {
     }
 
 
+    /* Constants */
+
+    /** EIP-712 domain separator name for Protocore. */
+    string public constant DOMAIN_SEPARATOR_NAME = "Mosaic-Core";
+
+    /** EIP-712 domain separator typehash for Protocore. */
+    bytes32 public constant DOMAIN_SEPARATOR_TYPEHASH = keccak256(
+        "EIP712Domain(string name,string version,bytes32 metachainId,address core)"
+    );
+
+    /** EIP-712 type hash for a Vote Message */
+    bytes32 public constant VOTE_MESSAGE_TYPEHASH = keccak256(
+        "VoteMessage(bytes32 transitionHash,bytes32 sourceBlockHash,bytes32 targetBlockHash,uint256 sourceBlockNumber,uint256 targetBlockNumber)"
+    );
+
+
     /* Storage */
 
     mapping(bytes32 /* vote message hash */ => Link) public links;
 
+    /** EIP-712 domain separator. */
+    bytes32 public domainSeparator;
+
     uint256 public openKernelHeight;
     bytes32 public openKernelHash;
 
 
+    /* Public Functions */
+
+    /**
+     * @notice Sets up Protocore contract.
+     *
+     * @param _metachainId Metachain Id.
+     * @param _core Core contract address.
+     */
+    function setup(
+        bytes32 _metachainId,
+        address _core
+    )
+        public
```

Comment:
Change this to internal, so that it can be only called from `OriginProtocore` and `SelfProtocore`.

Reply:
Done